portable/GCC/ARM_CM7/r0p1/port.c

```diff
@@ -828,9 +827,7 @@ static void vPortEnableVFP( void )
     __asm volatile
     (
         "   ldr.w r0, =0xE000ED88       \n" /* The FPU enable bits are in the CPACR. */
-        "   ldr r1, [r0]                \n"
-        "                               \n"
-        "   orr r1, r1, #( 0xf << 20 )  \n" /* Enable CP10 and CP11 coprocessors, then save back. */
+        "   mov.w r1, #( 0xf << 20 )    \n" /* Enable CP10 and CP11 coprocessors, and save. */
```

Comment:
Before the code was loading the value of the CPACR and then setting bits 23->20. This new code looks to directly assign register R1 to `0x00F00000`. Wouldn't this be dropping the [access privileges for the coprocessor](https://developer.arm.com/documentation/dui0646/c/Cortex-M7-Peripherals/Floating-Point-Unit/Coprocessor-Access-Control-Register)?

Reply:
I don't understand. Before it loaded CPACR into R1, set the bits, and store it again into CPACR. This skips the load, since all other bits are unused. (And according to my CM 7 programming manual from ST 0.)

Reply:
You're right! I should really read the docs which I link better